frontend/src/metabase/containers/ItemPicker/ItemPicker.unit.spec.js

```diff
@@ -294,6 +307,45 @@ describe("ItemPicker", () => {
       expect(items[1]).toHaveTextContent(COLLECTION.REGULAR_2.name);
       expect(items[2]).toHaveTextContent(COLLECTION.REGULAR.name);
     });
+
+    it("should filter collections", async () => {
+      await setup({
+        query: "foo",
+        collectionFilter: (collection, _index, allCollections) =>
+          !isPersonalCollectionOrChild(collection, allCollections) ||
+          collection.id === "root",
+      });
+
+      expect(screen.queryByText(/personal/i)).not.toBeInTheDocument();
+    });
+
+    it("should show search results", async () => {
+      await setup();
+
+      userEvent.click(screen.getByRole("img", { name: /search/ }));
+      userEvent.type(screen.getByPlaceholderText("Search"), "das{enter}");
+
+      expect(
+        await screen.findByText(/^regular dashboard$/i),
+      ).toBeInTheDocument();
+      expect(await screen.findByText(/nested/i)).toBeInTheDocument();
+      expect(await screen.findByText(/personal/i)).toBeInTheDocument();
+    });
+
+    it("should not show items of filtered collections when searching", async () => {
+      await setup({
+        collectionFilter: collection =>
+          collection.personal_owner_id === null || collection.id === "root",
```

Comment:
Should this `collectionFilter` be the same as lines +315 to +316?

```suggestion
        collectionFilter: (collection, _index, allCollections) =>
          !isPersonalCollectionOrChild(collection, allCollections)
```